### PR TITLE
Files watcher exclude - do not ignore changes to the node_modules folder itself.

### LINKED ISSUE #125886: macOS: Changes to node_modules folder are ignored
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.56.2
- OS Version: macOS BigSur 11.2.3

Steps to Reproduce:

1. Open Node.js project
2. Execute `npm install` in terminal
3. `node_modules` is not shown in Explorer side bar.

node

It seems to be by designed for performance reasons #23954

> Actually there is a difference between Windows and macOS/Linux: we ignore **/node_modules/** but that should actually not ignore changes to the node_modules folder itself. The real issue here is that our glob library is matching node_modules for a pattern of **/node_modules/** when it should not.

For now, `files.watcherExclude` has different default value for Windows and macOS/Linux.

Windows
```json
"**/node_modules/*/**": true
```

macOS/Linux
```json
"**/node_modules/**": true,
```

I recommend using the new default setting on macOS/Linux
```json
"**/node_modules/**/*": true
```

node-project-1

### PR DESCRIPTION
**Pull request by Aaaaash**
This PR fixes #125886

[Glob online tester](https://codepen.io/aaaaash/pen/qBrypvB)

node_modules-watcger

### CODE REVIEW

**Comment by Aaaaash** [comment]
[same issue](https://github.com/microsoft/vscode/issues/23954#issuecomment-292074903)

**Comment by Aaaaash** [comment]
Friendly ping @bpasero

**Comment by bpasero** [comment]
What issue does this address?

**Comment by Aaaaash** [comment]
updated

File: src/vs/workbench/contrib/files/browser/files.contribution.ts [hunk 1]
```diff
@@ -239,7 +239,7 @@ configurationRegistry.registerConfiguration({
 		},
 		'files.watcherExclude': {
 			'type': 'object',
-			'default': isWindows /* https://github.com/microsoft/vscode/issues/23954 */ ? { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/*/**': true, '**/.hg/store/**': true } : { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/**': true, '**/.hg/store/**': true },
+			'default': isWindows /* https://github.com/microsoft/vscode/issues/23954 */ ? { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/*/**': true, '**/.hg/store/**': true } : { '**/.git/objects/**': true, '**/.git/subtree-cache/**': true, '**/node_modules/**/*': true, '**/.hg/store/**': true },
```
Comment (bpasero) [inline src/vs/workbench/contrib/files/browser/files.contribution.ts:242]: @Aaaaash any reason we wouldn't just pick the exact same pattern as on Windows? From my testing, the advantage of `'**/node_modules/*/**'` over `'**/node_modules/**/*'` seems to be that you would see changes to top level folders in the `node_modules` folder, but nothing deeper. So essentially the explorer refreshes nicely on operations such as `yarn add <module>`. 

With your suggested pattern, I only see `node_modules` folder appearing and hiding, but not changes inside the folder.

Comment (bpasero) [reply src/vs/workbench/contrib/files/browser/files.contribution.ts:242]: Ping @Aaaaash

Comment (Aaaaash) [reply src/vs/workbench/contrib/files/browser/files.contribution.ts:242]: Sorry for the late reply, you are right, so we can use the same pattern on Windows and Linux/macOS ？

Comment (bpasero) [reply src/vs/workbench/contrib/files/browser/files.contribution.ts:242]: Yeah I think so, can you update the PR?

Comment (Aaaaash) [reply src/vs/workbench/contrib/files/browser/files.contribution.ts]: Updated

**Comment by bpasero** [comment]
Thanks 👍

**Pull request merged**